### PR TITLE
Add a Parquet file with column chunk key-value metadata

### PR DESCRIPTION
**Pull request by clee704**
This file has a single row group with 0 row and 2 columns.

The first column chunk has the following key-value metadata:
- The first entry has a key "foo" mapped to a value "bar".
- The second entry has a key "thisiskeywithoutvalue" and does not have a value.

Created with this code, with a modified Arrow library to output "thisiskeywithoutvalue" without a value for the first column.

```c++
PARQUET_ASSIGN_OR_THROW(auto sink,
                        arrow::io::FileOutputStream::Open(
                            "column_chunk_key_value_metadata.parquet"));
auto writer = parquet::ParquetFileWriter::Open(
    sink, std::static_pointer_cast<parquet::schema::GroupNode>(
              parquet::schema::GroupNode::Make(
                  "schema", parquet::Repetition::REQUIRED,
                  {parquet::schema::PrimitiveNode::Make(
                       "column1", parquet::Repetition::OPTIONAL,
                       parquet::Type::INT32),
                   parquet::schema::PrimitiveNode::Make(
                       "column2", parquet::Repetition::OPTIONAL,
                       parquet::Type::INT32)})));
auto rg_writer = writer->AppendRowGroup();
rg_writer->NextColumn()->key_value_metadata().Append("foo", "bar");
rg_writer->NextColumn();
```

This is for https://github.com/apache/arrow/pull/41580

### CODE REVIEW

**Comment by wgtmac** [comment]
Would you mind adding some description about this new file in the README?

**Comment by pitrou** [comment]
Suggestion: add two metadata entries: one with a value, the other without (the metadata key is mandatory while the value is [optional](https://github.com/apache/parquet-format/blob/e91ab5e892391bd179d3c16b7c1cf4fbaeebbfe7/src/main/thrift/parquet.thrift#L745-L748)).

**Review by mapleFU — APPROVED** [review]

File: data/README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@
 | concatenated_gzip_members.parquet     | 513 UINT64 numbers compressed using 2 concatenated gzip members in a single data page |
 | byte_stream_split.zstd.parquet | Standard normals with `BYTE_STREAM_SPLIT` encoding. See [note](#byte-stream-split) below |
 | incorrect_map_schema.parquet | Contains a Map schema without explicitly required keys, produced by Presto. See [note](#incorrect-map-schema) |
+| column_chunk_key_value_metadata.parquet | two INT32 columns, one with column chunk key-value metadata {"foo": "bar"} |
```
Comment (mapleFU) [inline data/README.md:54]:
```suggestion
| column_chunk_key_value_metadata.parquet | two INT32 columns, one with column chunk key-value metadata {"foo": "bar", "thisiskeywithoutvalue": ""} |
```

Comment (clee704) [reply data/README.md:54]: thanks for the correction!

**Comment by mapleFU** [comment]
Merged, thanks!

**Pull request merged**